### PR TITLE
cpu/esp8266: fix pwm_set func

### PR DESCRIPTION
**Pull request by gschorcht**
### Contribution description

In `pwm_set` function, switch-on and switch-off times for PWM channels were only determined for the following phase, but not for the current phase. This could result in

- a missing duty cycle when function `pwm_set` was called before switch-on time of the current phase or 
- an extended duty cycle function `pwm_set` was called before switch-off time of the current phase.

This PR fixes the problem.

### Testing procedure

Use `test/periph_pwm` with following changes in `main.c`:

```diff
-#define OSC_STEP        (10)
+#define OSC_STEP        (1)
-#define OSC_STEPS       (1000U)
+#define OSC_STEPS       (100U)
```
Flash the application with and without the changes in this PR to compare the results.
```
make BOARD=esp8266-esp-12x -C tests/periph_pwm flash test
```
Execute the `osci` command. Without the changes of this PR, you should be able to observe one extended duty cycle every 9th cycles. That is, `pwm_set` function is executed in 10th cycle and does not switch off the output port.

snapshot_2019-02-10_16-04-39￼

With changes of this PR, the output port should be switched on and off also in 10th cycle.

### CODE REVIEW

**Comment by MrKevinWeiss** [comment]
> make BOARD=esp8266-esp-12x -C tests/periph_pwm flash test

For your testing procedure you should maybe clarify use term not test (it has happened in a few PRs)

File: cpu/esp8266/periph/pwm.c [hunk 1]
```diff
@@ -183,23 +183,31 @@ void pwm_set(pwm_t pwm, uint8_t channel, uint16_t value)
...
         case PWM_CENTER:
-            _pwm_dev.chn[channel].next_on  = phase + (_pwm_dev.res - value) / 2;
+            next_on = phase + (_pwm_dev.res - value) / 2;
             break;
     }
 
-    _pwm_dev.chn[channel].next_off = _pwm_dev.chn[channel].next_on + value;
+    next_off = next_on + value;
+
+    next_on = (_pwm_dev.cycles < next_on) ? next_on : next_on + _pwm_dev.res;
```
Comment (MrKevinWeiss) [inline cpu/esp8266/periph/pwm.c]: I find 
```c
if (_pwm_dev.cycles >= next_on) {
    next_on += _pwm_dev.res;
}
```
a bit easier to read but I won't push, just a suggestion (even though it is more lines)

Comment (gschorcht) [reply cpu/esp8266/periph/pwm.c]: @MrKevinWeiss Finally I changed it as suggested by you.

**Comment by MrKevinWeiss** [comment]
Thanks, I will have it tested by tomorrow afternoon

File: cpu/esp8266/periph/pwm.c [hunk 2]
```diff
@@ -203,8 +203,12 @@ void pwm_set(pwm_t pwm, uint8_t channel, uint16_t value)
 
     next_off = next_on + value;
 
-    next_on = (_pwm_dev.cycles < next_on) ? next_on : next_on + _pwm_dev.res;
-    next_off = (_pwm_dev.cycles < next_off) ? next_off : next_off + _pwm_dev.res;
+    if (_pwm_dev.cycles < next_on) {
```
Comment (MrKevinWeiss) [inline cpu/esp8266/periph/pwm.c]: Maybe this should be `if (_pwm_dev.cycles >= next_on) {`

File: cpu/esp8266/periph/pwm.c [hunk 3]
```diff
@@ -203,8 +203,12 @@ void pwm_set(pwm_t pwm, uint8_t channel, uint16_t value)
...
+        next_on = next_on + _pwm_dev.res;
+    }
+    if (_pwm_dev.cycles < next_off) {
```
Comment (MrKevinWeiss) [inline cpu/esp8266/periph/pwm.c]: also this `if (_pwm_dev.cycles >= next_off) {`

File: cpu/esp8266/periph/pwm.c (hunk 2 above)
Comment (gschorcht) [reply cpu/esp8266/periph/pwm.c]: :confused: Of course, sorry changed it between two other tasks. Fixed.

File: cpu/esp8266/periph/pwm.c (hunk 3 above)
Comment (gschorcht) [reply cpu/esp8266/periph/pwm.c]: Fixed

**Comment by gschorcht** [comment]
@MrKevinWeiss Thanks for reviewing and testing.

**Pull request merged**